components/nimbus/src/enrollment.rs

```diff
@@ -270,6 +275,26 @@ impl ExperimentEnrollment {
                             branch: branch.clone(),
                         },
                     }
+                } else if updated_experiment.is_rollout
+                    && matches!(
+                        self.status,
+                        EnrollmentStatus::Disqualified {
+                            reason: DisqualifiedReason::NotSelected
+                                | DisqualifiedReason::NotTargeted,
```

Comment:
This is confusing; you're matching on `self.status` and you've already matched on `self.status` (to get into this arm of the match).

Consider adding `reason` to the [outer match arm](https://github.com/mozilla/application-services/pull/5716/files#diff-5be1ba032964b639be01987de468be59027bf9f057f8907b4c324a749d465080R260-R264), and matching on that.

Reply:
🤦 Idk how I missed that, great callout